### PR TITLE
Issue 813: Add support for default-group in ACI mode

### PR DESCRIPTION
**Pull request by srampal**
@joji PTAL

### CODE REVIEW

**Comment by jojimt** [comment]
As discussed, please add a check for default-net and add the new test to Makefile.

**Comment by srampal** [comment]
@Joji  Added the test to Makefile … however had to retain the logic of not depending on default-net so that the system tests (runContainer function in docker_test.go and so on) would not need to be refactored to become ACI aware. We can discuss more details offline if needed.

Re-submited PR with latest changes PTAL

File: test/systemtests/docker_test.go [hunk 1]
```diff
@@ -67,6 +67,12 @@ func (d *docker) runContainer(spec containerSpec) (*container, error) {
 		netstr = "--net=" + netstr
 	}
 
+        // Handle the case of "default-group" not specified explicitly 
+        // but only for the aci case (detected here via special aci network name)
+        if (spec.serviceName == "") && (spec.networkName == "default-net-aci"){
+                netstr = "--net=default-group"
```
Comment (jojimt) [inline test/systemtests/docker_test.go]: Why do we have this? This defeats the purpose of the test. The whole point is that in ACI mode, the code you added takes care of this automatically.

File: test/systemtests/aci_test.go [hunk 2]
```diff
@@ -87,6 +87,94 @@ func (s *systemtestSuite) TestACIMode(c *C) {
...
+        containersDefault := append( containersA, containersC...)
+
+	// Verify containers within the combined default-group can ping each other 
+	c.Assert(s.pingTest(containersDefault), IsNil)
+	// Verify containers within epgb can ping each other
+	c.Assert(s.pingTest(containersB), IsNil)
+	// Verify containers within the combined default-group can't ping epgb containers
+	c.Assert(s.pingFailureTest(containersDefault, containersB), IsNil)
+
+	log.Infof("Triggering netplugin restart")
```
Comment (jojimt) [inline test/systemtests/aci_test.go]: There is no need to do netplugin restart test specifically for this. That's already covered separately. Please remove it from here.

**Comment by srampal** [comment]
@joji  made the changes you suggested  PTAL
Removed the netplugin restart case
Moved the system test case for this to k8s (make k8s-test)

**Comment by jojimt** [comment]
@srampal Please make sure tests pass.

**Comment by srampal** [comment]
Done @joji  PTAL

File: netmaster/master/endpoint.go [hunk 3]
```diff
@@ -118,6 +118,13 @@ func CreateEndpoint(stateDriver core.StateDriver, nwCfg *mastercfg.CfgNetworkSta
 	epCfg.ServiceName = ep.ServiceName
 	epCfg.EPCommonName = epReq.EPCommonName
 
+	// In ACI mode, if a pod does not have a group label, we will assume "default-group"
+	isAci, _ := IsAciConfigured()
+
+	if isAci && (len(epCfg.ServiceName) == 0) {
+		epCfg.ServiceName = "default-group"
```
Comment (jojimt) [inline netmaster/master/endpoint.go]: Please add a log.Infof here with a message we are overriding withe default-group.

**Pull request merged**